### ISSUE TITLE
Videos: Add Live Photos to Sub-Navigation

### ISSUE THREAD

**Issue by simoncoul** [photoprism/photoprism#1761]
Currently videos up to 3 seconds in duration are considered a Live/Motion Photo, regardless if they are actually a Live/Photo Photo or just a short video clip. This results in short video files being incorrectly categorized and not being present in the Videos Tab of the UI.

Apple, Google, and Samsung all use different formats for Live/Motion Photos; however they all achieve a final result of having images and videos with an image container (HEIC, JPEG, etc). Therefore a piece of media should only be treated as a Live/Motion Photo if it meets this criteria. Any video files within standard video containers (mov, mp4, etc) should be treated as a video regardless of duration.

Alternatively any video container could be treated as it currently is, but also show up in the Videos tab. The former solution is preferred as Videos are Live/Motion Photos convey different information (ie Baby's first words is only a couple second video, where a live photo is used to ensure baby is not blinking in a photo).

**Comment by lastzero** [comment]
The change will include live photos in video search results. It also adds a "Live" navigation item incl count for finding live photos / short videos only. Will be part of the next Development Preview build later today. Hope that solves your issue for now.

![live](https://user-images.githubusercontent.com/301686/144042383-43ca5304-bed0-4053-b516-fe7af9bd4595.jpg)

**Comment by simoncoul** [comment]
Unfortunately including live photos into the video search results will make this issue worse. Currently only short videos are excluded from the Video search, but iOS has Live Photos turned on by default since the iPhone 7. Therefore as time progresses the Videos search will become less useful as it will quickly become consumed by live photos instead of videos.

The videos search should be restricted to media uploaded within a video container, and not videos extracted from the originally uploaded media.

**Comment by lastzero** [comment]
These are the test files we got from an Apple iPhone user:

https://dl.photoprism.org/samples/iPhone%2520X/

Two files per Live Photo, one image, one mov *video*:

- `IMG_3982.heic`
- `IMG_3982.mov`

We can't change it back and forth all the time as there are other issues that need attention as well. We're trying to do you a favor here. Don't even know if you're a sponsor...

**Comment by simoncoul** [comment]
I appreciate the support in addressing this. I guess the issue at hand is the context in which the video is added to the library. For a live photo I upload a single heic (iPhone) file and photoprism processes it and results in the original, thumbnail, sidecar, and mov files.

![Screenshot_20211130-090604~2](https://user-images.githubusercontent.com/6271217/144062840-a90a020c-652c-4750-bbb3-62edcf304f60.png)

When I upload a video it processes and results in the original video and thumbnail files.

![Screenshot_20211130-090800~2](https://user-images.githubusercontent.com/6271217/144063256-a5d2be3a-e98b-4f45-8071-ffe928d12bf8.png)

The ideal solution would be a video file should be flagged, either that it was uploaded as a video or that it was extracted. Alternative a check could be run that if a specific set of conditions is met (heic+mov+xmp with same file name) it is a live photo.

**Comment by lastzero** [comment]
Too big a change for a small update. We can look at this next year and should also talk to other users as they are the reason for the current implementation.

Live Photos have been moved to the Videos sub-navigation, so you can easily find them there if needed. Video results don't include Live Photos anymore - as requested.

Note you can change the type of a Live Photo to "Video" in the photo edit dialog (we know batch editing would be helpful - it's on our ToDo list and Roadmap already). Advanced users can alternatively change the type with a little SQL in the database for all files they want to see in videos instead (`photos.photo_type`).

![live-video](https://user-images.githubusercontent.com/301686/144064865-7604815e-0edf-45dd-b72e-64b46c8dd7d8.jpg)

**Comment by simoncoul** [comment]
Thanks again for the support!

The below command in search should identify most non-iphone videos that have misidentified as a live photo. This produced a few hundred items that I could manual change with the method you mentioned.

filename:"*.avi | *.m4v | *.mp4 | *.mpg | *.mpeg | *.3gp" | type:live

This still does not account for short iphone videos which I believe could be identified by aspect ratio if it was a searchable field  (1.33/0.75 instead of 1.79/0.56). Then something like the below command would work:

filename:"*.mov" | aspectratio:"1.79 | 0.56"

**Comment by GlassedSilver** [comment]
A Live Photo is a photo, not a video. I don't know a single phone OEM showing you Live Photo equivalents as videos.

The video part is to make the photo more lively, it's technically a cross-medium, but it's always thought of as (and composed like) a photo. At least in my humble experience.

Not sure if saying something like "Don't even know if you're a sponsor..." is very helpful either.

From a sponsor: if I see a cool project and I notice it's a good candidate for a software deployment and I'm providing feedback that's because a) I'm showing a commitment to the development of the application and b) do you really think this will convince the average user to sponsor if they feel guilt-tripped into it?

I see where you're coming from logically, it takes resources to develop this application and that's appreciated. By both paying as well as non-paying users. To go around and drop sentences nonchalantly like that however you risk a part of your reputation.

tl;dr: another way to look at feedback is to take it as someone taking time out of their day to share feedback.

**Comment by lastzero** [comment]
Thanks for your feedback!
